### ISSUE TITLE
Adding a resume session feature

### ISSUE THREAD

**Issue by rondiplomatico** [acroz/pylivy#67]
Hey, so i've been using pylivy quite a bit for a recent project and found that it greatly eases development if you can specify a session id for an already running session. This is especially handy if
the spark session you are connecting to takes quite a while to set up and load data.

To get an idea of what's been done, checkout my branch https://github.com/rondiplomatico/pylivy/tree/feature/resume i'd also create a PR from if you'd like to include this.

Cheers

**Comment by acroz** [comment]
This is a great suggestion, though I note that in the case we're resuming, most of the constructor arguments are unused.

What would you think about changing the interface to have two public interfaces for building a LivySession, implemented through classmethods, e.g.:

```python
# In the case you want a new session:
session = LivySession.new(url, kind=SessionKind.SPARK)
# In the case you want an existing session
session = LivySession.connect(url, session_id=5)
```

What do you think?

Separately, the current implementation closes the session by default when done when used as a context manager. Do you think that behaviour should not exist in this case? Or would you just not use it as a context manager?

**Comment by rondiplomatico** [comment]
Using two interfaces is fine with me and makes the matter even clearer. However, is adding a "new"-method in addition to the current way a reasonable change to the status quo (As people will probably not want to change all their code)? So i'd suggest using

```python
# In the case you want a new session:
session = LivySession(url, kind=SessionKind.SPARK, <args>...)
# In the case you want an existing session:
session = LivySession.resume(url, session_id=5)
```
This will also keep everything more intuitive regarding the context manager. For the "resume" case, i'd propose to add an internal flag `resumed=True` and act in one of the two ways:
a) Simply not close the session on `__exit__`. Additionally, spit out an info/warning "Resuming a session in conjunction with Python's context management will not terminate the session"
b) As soon as the `__enter__` method is invoked and `resumed==True`, raise an Exception and state that using pythons context management feature in conjunction with resuming a session is not admitted by design. why have a context where there is no resource/connection/filepointer etc to clean up afterwards..

Cheers!

**Comment by acroz** [comment]
Hi @rondiplomatico, I finally got round to looking into this again - could you have a look and give any thoughts on #76?

**Comment by acroz** [comment]
This is now resolved in [0.7.0](https://pypi.org/project/livy/0.7.0/). To create a new session, call `LivySession.create()`, and to connect to an existing one construct `LivySession` directly.
